src/main/scala/mesosphere/marathon/api/v2/json/Formats.scala

```diff
@@ -993,7 +993,7 @@ trait AppAndGroupFormats {
       // top-level ports fields are incompatible with IP/CT
       if (runSpec.ipAddress.isEmpty) {
         appJson = appJson ++ Json.obj(
-          "ports" -> runSpec.servicePorts,
+          "ports" -> runSpec.portDefinitions.map(_.port),
```

Comment:
Until every port is assigned a `servicePort`, they're not explicitly opt-in, and making them so would be a breaking change.